doc/api/next_api_changes/deprecations/23565-OG.rst

```diff
@@ -0,0 +1,7 @@
+Alternative import location for ``TextPath`` and ``AxislineStyle``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These two classes have historically sometimes been imported from incorrect
+locations. This is now deprecated and `.TextPath` should be imported from
```

Comment:
I'm not sure we should do this. If we deprecate we break a lot of user code eventually.

https://github.com/search?q=%22from+matplotlib.textpath+import+TextPath%22&type=code
textpath -> 825 hits

https://github.com/search?q=%22from+matplotlib.text+import+TextPath%22&type=code
text -> 1783 hits

Could do a pending deprecation.

I also don't know which way I would deprecate. Sure, TextPath is in TextPath (alongside with TextToPath). But is it really necessary to have that many modules as API surface. I can see an argument that these are text-like enough to be accessed via the `text` module. - Which would long-term mean making `textpath` private, or moving the contents to text directly.

Reply:
Agreed that if we really want to pick one module (which is not clear), I'd rather make text the "official" one.

Reply:
OK! But then maybe we should at least make sure that `TextPath` is documented as being in `text` and not in `textpath`? https://matplotlib.org/devdocs/api/textpath_api.html

Reply:
`textpath` was made its own module in 2009. Although it doesn't really mean that it was fullt used for its purposes.

Reply:
See #23576 for the `textpath` module.

Reply:
As a preparation for the dev call:

`textpath` 25 hits (5 matplotlib, 8 Visualization book)
https://grep.app/search?q=from%20matplotlib.textpath%20import%20TextPath

`text` 13 hits (3 matplotlib, 3 matplotlib-cn, 4 Visualization book)
https://grep.app/search?q=from%20matplotlib.text%20import%20TextPath